code/game/machinery/big_manipulator.dm

```diff
@@ -0,0 +1,261 @@
+	. = ..()
+	if(default_deconstruction_crowbar(tool))
+		return ITEM_INTERACT_SUCCESS
+	return ITEM_INTERACT_BLOCKING
+
+/obj/machinery/big_manipulator/RefreshParts()
+	. = ..()
+
+	manipulator_lvl()
+
+/// Creat manipulator hand effect on manipulator core.
+/obj/machinery/big_manipulator/proc/create_manipulator_hand()
+	manipulator_hand = new/obj/effect/big_manipulator_hand(get_turf(src))
+	manipulator_hand.dir = take_here
+
+/// Check servo tier and change manipulator speed, power_use and colour.
+/obj/machinery/big_manipulator/proc/manipulator_lvl()
+	var/datum/stock_part/servo/locate_servo = locate() in component_parts
+	if(!locate_servo)
+		return
+	switch(locate_servo.tier)
+		if(1)
+			working_speed = 2 SECONDS
+			power_use_lvl = 0.2
+			set_greyscale(COLOR_YELLOW)
+			manipulator_hand?.set_greyscale(COLOR_YELLOW)
+		if(2)
+			working_speed = 1.4 SECONDS
+			power_use_lvl = 0.4
+			set_greyscale(COLOR_ORANGE)
+			manipulator_hand?.set_greyscale(COLOR_ORANGE)
+		if(3)
+			working_speed = 0.8 SECONDS
+			power_use_lvl = 0.6
+			set_greyscale(COLOR_RED)
+			manipulator_hand?.set_greyscale(COLOR_RED)
+		if(4)
+			working_speed = 0.2 SECONDS
+			power_use_lvl = 0.8
+			set_greyscale(COLOR_PURPLE)
+			manipulator_hand?.set_greyscale(COLOR_PURPLE)
+
+	active_power_usage = BASE_MACHINE_ACTIVE_CONSUMPTION * power_use_lvl
+
+/// Changing take and drop turf tiles when we anchore manipulator.
+/obj/machinery/big_manipulator/proc/anchored_check()
+	if(anchored)
+		take_turf = get_step(src, take_here)
+		drop_turf = get_step(src, drop_here)
+	else
+		take_turf = null
+		drop_turf = null
+
+/// Changing take and drop turf dirs and also changing manipulator hand sprite dir.
+/obj/machinery/big_manipulator/proc/rotate_big_hand()
+	switch(take_here)
+		if(NORTH)
+			take_here = EAST
+			drop_here = WEST
+		if(EAST)
+			take_here = SOUTH
+			drop_here = NORTH
+		if(SOUTH)
+			take_here = WEST
+			drop_here = EAST
+		if(WEST)
+			take_here = NORTH
+			drop_here = SOUTH
+	manipulator_hand.dir = take_here
+	anchored_check()
+
+/// Deliting hand will destroy our manipulator core.
+/obj/machinery/big_manipulator/proc/on_hand_qdel()
+	SIGNAL_HANDLER
+
+	deconstruct(TRUE)
+
+/// Pre take and drop proc from [take and drop procs loop]:
+/// Check if we have item on take_turf to start take and drop loop
+/obj/machinery/big_manipulator/proc/is_work_check()
+	for(var/obj/item/take_item in take_turf.contents)
+		try_take_thing(take_turf, take_item)
+		break
+
+/// First take and drop proc from [take and drop procs loop]:
+/// Check if we can take item from take_turf to work with him. This proc also calling from ATOM_ENTERED signal.
+/obj/machinery/big_manipulator/proc/try_take_thing(datum/source, atom/movable/target)
+	SIGNAL_HANDLER
+
+	if(!on)
+		return
+	if(!anchored)
+		return
+	if(QDELETED(source) || QDELETED(target))
+		return
+	if(on_work)
+		return
+	if(!use_energy(active_power_usage, force = FALSE))
+		on = !on
```

Comment:
```suggestion
		on = FALSE
```
You don't want to toggle, just disable here. It doesnt matter but it helps with readability